src/contracts/VouchRegister.sol

```diff
@@ -3,26 +3,27 @@ pragma solidity ^0.8;
 
 contract VouchRegister {
     event Vouch(
-        address solver,
-        address bondingPool,
+        address indexed solver,
+        address indexed bondingPool,
         address cowRewardTarget,
-        address txSender
+        address indexed txSender
```

Comment:
naming-nit: `txSender` is a bit confusing to me as `tx.origin`. Maybe `sender` is enough?